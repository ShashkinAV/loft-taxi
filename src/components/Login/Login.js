import React from 'react';

const Login = ({setRoute}) => {
	return (
	  <>
		<h1>Login</h1>
		<form onSubmit={()=>setRoute("profile")}>
		  <label>
			Email:
			<input name="email" type="email"/>
		  </label>
		  <label>
			Password:
			<input name="password" type="password"/>
		  </label>
		  <input type="submit" value="Login" />
		</form>
	  </>
	);
  }

export default Login; 