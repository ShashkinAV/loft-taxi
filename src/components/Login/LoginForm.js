import React from 'react';
import { Link } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { getAuth, fetchAuthRequest } from '../../modules/main';
import {shallowEqual, useSelector, useDispatch} from 'react-redux';
import history from '../../history';

const useStyles = makeStyles(theme => ({
  card: {
    padding: 50,
    minWidth: 400,
    marginTop: 48,
    marginBottom: 48,
  },
  title: {
    fontSize: 36,
    fontWeight: 400,
    marginBottom: 30
  },
  btnGrid: {
    marginTop: 30
  }
}));

export const LoginForm = () => {
  const classes = useStyles();

  const auth = useSelector(getAuth, shallowEqual);
  const authAction = useSelector(fetchAuthRequest, shallowEqual);

  const dispatch = useDispatch();

  if (auth && auth.success && JSON.parse(auth.success) === true) {
    localStorage.setItem('authSuccess', auth.success);
    localStorage.setItem('authToken', auth.token);
    history.push('/map');
  }

  const onSubmit = (e) => {
    e.preventDefault();

    dispatch({
      ...authAction,
      payload: {
          email: e.target.email.value,
          password: e.target.password.value
      }
    });

    // console.log(auth);
    // console.log(authAction);
   }

  return (
    <Card className={classes.card}>
            <form  onSubmit={onSubmit}>
              <Grid container>
                <Grid item xs={12}>
                  <Typography variant='h4' component='h1'>
                    Войти
                  </Typography>
                </Grid>
                <Grid item xs={12}>
                  <p>
                    Новый пользователь?{' '}
                    <Link to={'/signup'} >
                      Зарегистрируйтесь
                    </Link>
                  </p>
                </Grid>
                <Grid item xs={12}>
                  <TextField fullWidth required name="email" type="email" label="Имя пользователя"/>
                </Grid>
                <Grid item xs={12} >
                  <TextField fullWidth required name="password" type="password" label="Пароль" margin="normal"/>
                </Grid>
                <Grid item xs={12} align="right" className={classes.btnGrid}>
                  <Button type="submit" variant="contained" color="primary">
                    Войти
                  </Button>
                </Grid>
              </Grid>
            </form>
    </Card>
  )
}