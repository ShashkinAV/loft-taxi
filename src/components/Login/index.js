import React from 'react';
import PropTypes from "prop-types";
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import {LoginForm} from './LoginForm';
import { Logo } from 'loft-taxi-mui-theme';

const BackgroundURL = '../background.jpg';

const useStyles = makeStyles(theme => ({
  login: {
    width: '100%',
    minHeight: '100vh',
    backgroundSize: 'cover',
    backgroundImage: `url(${BackgroundURL})`,
    flexGrow: 1,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  }
}));

export const Login = () => {
  const classes = useStyles();

  return (
    <div className={classes.login}>
       <Grid 
        container
        direction="row"
        justify="center"
        alignItems="center">
          <Grid item xs={3}>
            <Logo white animated />
          </Grid>
          <Grid item xs={3}>
            <LoginForm/>
          </Grid>
       </Grid>
    </div>
  );
}

Login.propTypes = {
  setRoute: PropTypes.func
};
