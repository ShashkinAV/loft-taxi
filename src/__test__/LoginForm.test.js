import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { LoginForm } from '../components/Login/LoginForm';
import renderer from 'react-test-renderer';

describe('LoginForm', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LoginForm/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders LoginForm', () => {
    const wrapper = shallow(<LoginForm/>);
    /*--TODO to fixit--*/
    // expect(wrapper.find('.LoginForm').length === 1).toEqual(false);
  });
});

// FIX TypeError: window.URL.createObjectURL is not a function
jest.mock('mapbox-gl', () => ({
  Map: () => ({})
}))