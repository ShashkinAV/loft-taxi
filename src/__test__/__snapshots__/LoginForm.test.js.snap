// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h1
          className="MuiTypography-root MuiTypography-h4"
        >
          Войти
        </h1>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p>
          Новый пользователь?
           
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="#"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            Зарегистрируйтесь
          </a>
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink={false}
          >
            Имя пользователя
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="email"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink={false}
          >
            Пароль
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
            />
          </div>
        </div>
      </div>
      <div
        align="right"
        className="MuiGrid-root makeStyles-btnGrid-3 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Войти
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
